### DISCUSSION TITLE
Why the older version has been removed & why a rework ?

### DISCUSSION THREAD

**Discussion by Stun3R** [Stun3R/strapi-sdk-js#32] [Announcements]
## 👋🏻 Welcome
First, thank you for the interest you put in this project!
As you may know, I already wrote an SDK for [Strapi](https://strapi.io) and it had the same name (`strapi-sdk-js`), same package name & same documentation link...

You may ask "why you didn't released a new version and let the older one?". 

### The answer is simple:
The first version doesn't suit me anymore in relation to the skills I had at the time I had to develop it. The project lacks readability, the code wasn't very clean (everything in one file). This project needed to be more professional, the documentation wasn't clear either and some features were missing...

**_That's why I decided to not released a V2 as I wanted to rework it from scratch._**

### What were the overall ideas ?
- A Typescript project that supports GraphQL, token management (localStorage or Cookies) & doesn't have a greedy & heavy HTTP client.
- It must handle basic methods as well as authentication.
- If possible a CLI that allows to generate GraphQL types/mutations found on the user's API

### What is it finally ?
**Strapi SDK** is a beautiful **Typescript** project which support **CRUD**, **GraphQL** & **authentication** methods. For the authentication part you can chose between the **local storage** & **cookies** in order to store your Strapi JWT.

Because this package has been designed using Typescript, it natively supports it. You have access to all the types used for the development and in all his methods.

The **CLI** part has been **removed** from the package since **it cost too much memory and was too heavy for a SDK**. You were forced to use `GraphQL` in your Strapi API in order to make it works... **So I decided to create another package for it which is [strapi-generate-types](https://www.npmjs.com/package/strapi-generate-types).**

I kept the **generation feature** in mind and I'm **working hard on it**! You'll have the ability to install a plugin on your Strapi API which will **expose your content types**. The SDK will **trigger on load** this plugin's endpoint in order to **generate shortcuts methods** like `strapi.articles.find()` and **Typescript's types**. 🚀

### What's next ?
As I said I'm working on the generation feature and currently maintaining the core features.
Don't hesitate to open a Discussion or an ISSUE in order to give some feedback! ❤️

### Useful links
- [✨ &nbsp;Release Notes](https://github.com/stun3r/strapi-generate-types/releases)
- [📚 &nbsp;Documentation](https://strapi-sdk-js.netlify.app)

**Comment by feremabraz** [comment]
Very cool work, @Stun3R. Very impressive! 🆒😎
